config/http_config.go

```diff
@@ -110,11 +110,24 @@ func (u *URL) UnmarshalYAML(unmarshal func(interface{}) error) error {
 // MarshalYAML implements the yaml.Marshaler interface for URLs.
 func (u URL) MarshalYAML() (interface{}, error) {
 	if u.URL != nil {
-		return u.String(), nil
+		return u.Redacted(), nil
 	}
 	return nil, nil
 }
 
+// Redacted returns the URL but replaces any password with "xxxxx".
+func (u URL) Redacted() string {
+	if u.URL != nil {
+		ru := *u.URL
+		if _, ok := ru.User.Password(); ok {
+			// We can not use secretToken because it would be escaped.
+			ru.User = url.UserPassword(ru.User.Username(), "xxxxx")
+		}
+		return ru.String()
+	}
+	return ""
```

Comment:
Look nicer, less indentation
```suggestion
	if u.URL == nil {
		return ""
	}

	ru := *u.URL
	if _, ok := ru.User.Password(); ok {
		// We can not use secretToken because it would be escaped.
		ru.User = url.UserPassword(ru.User.Username(), "xxxxx")
	}
	return ru.String()
```